lib/TableGen/Operations.cpp

```diff
@@ -162,10 +163,18 @@ void OperationBase::emitArgumentAccessorDeclarations(llvm::raw_ostream &out,
     const bool isVarArg = arg.type->isVarArgList();
     std::string defaultDeclaration = "$0 get$1() $2;";
 
-    if (!isVarArg && !arg.type->isImmutable()) {
-      defaultDeclaration += R"(
-        void set$1($0 $3);
-      )";
+    if (!arg.type->isImmutable()) {
+      if (!isVarArg) {
+        defaultDeclaration += R"(
+          void set$1($0 $3);
+        )";
+      } else {
+        defaultDeclaration += R"(
+          /// Returns a new op with the same arguments and a new tail argument list.
+          /// The object on which this is called will be invalidated.
+          $_op *replace$1AndInvalidate(::llvm_dialects::Builder &, ::llvm::ArrayRef<Value *>);
```

Comment:
I think this function should not take a builder and instead create the CallInst directly.
For a setter like this, it’s more convenient if one does not need to pass a builder.

Reply:
I'm not certain on that, see the comment above. I am not sure we shouldn't pass a builder that guards and re-sets the insertion point.

Reply:
I think using CallInst::Create is fine in this case